ml-data-hub-plugin/src/main/groovy/com/marklogic/gradle/DataHubPlugin.groovy

```diff
@@ -140,7 +140,7 @@ class DataHubPlugin implements Plugin<Project> {
         WatchTask watchTask = project.tasks.getByName("mlWatch")
         CommandContext commandContext = new CommandContext(hubConfig.getAppConfig(), hubConfig.getManageClient(), hubConfig.getAdminManager())
         Versions versions = ((ApplicationContext)project.property("dataHubApplicationContext")).getBean(Versions.class)
-        watchTask.onModulesLoaded = new ModuleWatchingConsumer(commandContext, new GenerateFunctionMetadataCommand(hubConfig.newModulesDbClient(), versions))
```

Comment:
1. Instead of these changes, can 'GenerateFunctionMetadataCommand'   instance be obtained from the bean and used ? Something like     
GenerateFunctionMetadataCommand generateFunctionMetadataCommand = ((ApplicationContext)project.property("dataHubApplicationContext")).getBean(GenerateFunctionMetadataCommand.class)